### ISSUE TITLE
There were some problems when training your own dataset

### ISSUE THREAD

**Issue by buggogogo** [VinAIResearch/ISBNet#21]
Hello author！
     Thank you very much for your work, it has benefited me a lot. I changed my own data with reference to stpls3d. And modified the model-related files for training. But the one where the semantic tag starts (either started from 0 or 1) cannot be rendered in training.The problem is shown in the picture.Is there any way you can solve this problem?

![微信图片_20230612194935](https://github.com/VinAIResearch/ISBNet/assets/131874936/141d1011-26f1-44ba-a8d6-fcc7aeba04b7)

**Comment by ngoductuanlhp** [comment]
`nan` results mean that there is no point with the GT label as 'stem'. Maybe you set the instance_label of 'stem' points to -100 during preprocessing data. If both stem and leaf are instance classes, you should set `semantic_classes=2` and `instance_classes=2` in the config file.

**Comment by huanghuang113** [comment]
> Hello author！ Thank you very much for your work, it has benefited me a lot. I changed my own data with reference to stpls3d. And modified the model-related files for training. But the one where the semantic tag starts (either started from 0 or 1) cannot be rendered in training.The problem is shown in the picture.Is there any way you can solve this problem?
> 
> ![微信图片_20230612194935](https://user-images.githubusercontent.com/131874936/245135898-141d1011-26f1-44ba-a8d6-fcc7aeba04b7.jpg)

Hello sir, can you please share your code about processing your own dataset, I'm having some problems and need to preprocess my dataset but I'm experiencing some difficulties

**Comment by huanghuang113** [comment]
> Hello author！ Thank you very much for your work, it has benefited me a lot. I changed my own data with reference to stpls3d. And modified the model-related files for training. But the one where the semantic tag starts (either started from 0 or 1) cannot be rendered in training.The problem is shown in the picture.Is there any way you can solve this problem?
> 
> ![微信图片_20230612194935](https://user-images.githubusercontent.com/131874936/245135898-141d1011-26f1-44ba-a8d6-fcc7aeba04b7.jpg)

Hi, can you please show me how your config file is configured, I am currently learning about this, can you please show me how your config is configured?

**Comment by wowluna** [comment]
> Hello author！ Thank you very much for your work, it has benefited me a lot. I changed my own data with reference to stpls3d. And modified the model-related files for training. But the one where the semantic tag starts (either started from 0 or 1) cannot be rendered in training.The problem is shown in the picture.Is there any way you can solve this problem?
> 
> ![微信图片_20230612194935](https://user-images.githubusercontent.com/131874936/245135898-141d1011-26f1-44ba-a8d6-fcc7aeba04b7.jpg)

Perhaps you have solved this problem? I met the same problem.
![image](https://github.com/VinAIResearch/ISBNet/assets/102451556/7c3cd7d5-7342-4470-863b-d835a76e969b)

**Comment by wowluna** [comment]
hi!When I modified the following parameters of "instance_eval.py"and "ibsnet.py", I got an evaluation.
![image](https://github.com/VinAIResearch/ISBNet/assets/102451556/55308782-1212-40ca-b963-0a8203445ded)
 @![image](https://github.com/VinAIResearch/ISBNet/assets/102451556/05b3d714-e810-43bc-9087-7f7d2311c5ac)
But I don't understand the parameter "gts_sem". Can you explain it for me?   Thanks!!  @ngoductuanlhp @buggogogo

**Comment by ngoductuanlhp** [comment]
Dear @wowluna ,
`nan` in evaluation means there is no gt instance of this class in the evaluation samples. Let me clarify these lines of code. In ScanNetV2, there are 20 categories for semantic segmentation and 18 categories for instance segmentation (we assume that the number of semantic classes is always greater or equal to the number of instance classes), thus the difference is 2. The transformation here is `gts_sem = gts_sem + (num_semantic - num_instance) + 1`. The purpose is to align the categories in groundtruth annotations with the predicted categories in our predictions.

**Comment by wowluna** [comment]
> this class in the evaluation samples. Let me clarify these lines of code. In ScanNetV2, there are 20 categories for semantic segmentation and 18 categories for instance segmentation (we assume that the number of semantic classes is always greater or equal to the number of instance classes), thus the difference is 2. The transformation here is `gts_sem = gts_sem + (num_semantic - num_instance) + 1`. The purpose is to align the categories in groundtruth annotations with the predicted categories in our predictions.

Thanks！I get it now.

**Comment by lifeiwen** [comment]
@wowluna hello， I have a question to ask you. I also have two categories, one background and one foreground (Particle), with many instances. But when I modify the semantic category and instance category in the configuration file like this:
![捕获](https://github.com/VinAIResearch/ISBNet/assets/49854196/12fff285-11d2-4c3b-9f7b-9f15f3386e3e)
I get an error when running the training code. How specifically do you modify the configuration parameters?
![image](https://github.com/VinAIResearch/ISBNet/assets/49854196/14b66800-cae8-45ae-adcb-10fdfbaf5400)
